src/components/panel/panel.spec.js

```diff
@@ -643,6 +643,227 @@ describe('$mdPanel', function() {
       expect(scrollMaskEl).not.toExist();
       expect($mdUtil.disableScrollAround).toHaveBeenCalled();
     });
+
+    describe('animation hooks: ', function() {
+      it('should call onDomAdded if provided when adding the panel to the DOM',
+          function() {
+            var onDomAdded = jasmine.createSpy();
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            attachPanel(config);
+
+            expect(onDomAdded).toHaveBeenCalled();
+            expect(PANEL_EL).toExist();
+          });
+
+      it('onDomAdded should return a promise and continue resolving the ' +
+          'rest of promise chain', function() {
+            var onDomAdded = jasmine.createSpy();
+            var config = angular.extend(
+                {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+            openPanel(config);
+
+            expect(onDomAdded).toHaveBeenCalled();
+            expect(PANEL_EL).toExist();
+            expect(PANEL_WRAPPER_CLASS).not.toHaveClass(HIDDEN_CLASS);
+          });
+
+      it('should onDomAdded reject, the panel should not be shown', function() {
+
+        var onDomAdded = function() {
+          return panelRef._$q.reject();
+        };
+
+        var config = angular.extend(
+            {'onDomAdded': onDomAdded}, DEFAULT_CONFIG);
+
+        panelRef = $mdPanel.create(config);
+
+        var openRejected = false;
+
+        panelRef.show = function() {
+          return panelRef._$q.reject();
+        };
+
+        panelRef.open().then(function() {}, function () {
+          openRejected = true;
+        });
+
+        $rootScope.$apply();
+
+        expect(openRejected).toBe(true);
+        expect(panelRef.isAttached).toEqual(true);
+        expect(panelRef._panelContainer).toHaveClass(HIDDEN_CLASS);
+      });
+
+      it('should call onOpenComplete if provided after adding the panel to the ' +
+          'DOM and animating', function() {
+            var onOpenComplete = jasmine.createSpy();
+            var config = angular.extend(
+                {'onOpenComplete': onOpenComplete}, DEFAULT_CONFIG);
+
+            openPanel(config);
+
+            expect(onOpenComplete).toHaveBeenCalled();
+          });
+
+      it('should call onRemoving if provided when removing the panel from the DOM',
```

Comment:
hidePanel is not removing it from the DOM, just hiding it. Please update this test description.